From 1f90112f00a0b831a24c38e242e6b8d191a262e7 Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Thu, 11 Feb 2021 20:15:02 +0300
Subject: [PATCH 01/11] plat: renesas: rcar: bl31: Enable RPC access if
 necessary

This enables RPC flash access before leaving BL31
in case RCAR_RPC_HYPERFLASH_LOCKED is disabled.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 plat/renesas/rcar/bl31_plat_setup.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/plat/renesas/rcar/bl31_plat_setup.c b/plat/renesas/rcar/bl31_plat_setup.c
index 8dd3a5fef..c1ec987e0 100644
--- a/plat/renesas/rcar/bl31_plat_setup.c
+++ b/plat/renesas/rcar/bl31_plat_setup.c
@@ -141,3 +141,13 @@ uint32_t bl31_plat_boot_mpidr_chk(void)
 	return rc;
 }
 
+void bl31_plat_runtime_setup(void)
+{
+#if (RCAR_RPC_HYPERFLASH_LOCKED == 0)
+	/* Enable non-secure access to the RPC HyperFlash region. */
+	mmio_write_32(0xee2000b8, 0x155);
+	mmio_write_32(0xee200000, mmio_read_32(0xee200000) & 0x7fffffff);
+#endif
+
+	console_switch_state(CONSOLE_FLAG_RUNTIME);
+}
-- 
2.25.1

